packages/account/src/Sections/Profile/LanguageSettings/language-settings.tsx

```diff
@@ -24,7 +24,7 @@ const LanguageSettings = ({ changeSelectedLanguage, current_language, isCurrentL
             initialValues={initial_values}
             onSubmit={async values => {
                 const { language_code } = values;
-                changeSelectedLanguage(language_code);
+                await changeSelectedLanguage(language_code);
```

Comment:
Do you need to add Async to `changeSelectedLanguage`?

Reply:
Yes. It is better to wait for the code execution inside `changeSelectedLanguage` i.e waiting for authorize, calling set_settings, and then setting the i18 language